File: 02839.py
N = int(input())

five = N // 5
rest = N - five * 5
complete = False
while five:
    rest = N - five * 5
    if rest % 3 == 0:
        print(five + rest // 3)
        complete = True
        break

    five -= 1

if not complete:
    if N % 3 == 0:
        print(N // 3)
    else: print(-1)

File: 02839.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;

	int five = N / 5;
	int rest = N - five * 5;
	bool complete = false;
	while (five) {
		rest = N - five * 5;
		if (rest % 3 == 0) {
			cout << five + rest / 3 << endl;
			complete = true;
			break;
		}

		five--;
	}

	if (!complete) {
		if (N % 3 == 0)
			cout << N / 3 << endl;
		else cout << -1 << endl;
	}
}
